lib/update-lockfile.js

```diff
@@ -16,12 +16,6 @@ const yarnFlags = {
   'optionalDependencies': '-O'
 }
 
-const setPrefixYarn = prefix => prefix === '~'
```

Comment:
not too familiar with yarn, but would another approach be making a clause for `^` here where `dependency.range` is picked over `dependency.version` with `--exact` or `--tilde`?

Reply:
It would certainly be possible to preserve the current behavior and only switch to `dependency.range` for the caret, but IMO that would unnecessarily increase complexity in comparison to just going the one single code path and passing the "sign" directly.

Reply:
so you’re saying `yarn add --tilde 2.0.0` and `yarn add ~2.0.0` are functionally equivalent? same for `--exact`?

Reply:
I'll double check yarn's behavior in all cases when I have time later on.

Reply:
In fact, it seems like both `--exact` and `--tilde` are ignored when an explicit version (after `@`) is given (as of yarn latest 1.2.1). The flags are only taken into account when using only the package name...

Reply:
I've verified that my fork works with tilde
https://github.com/jeysal/ngx-input-flow/commit/4951160236d7bbed19f4cbfa4446ecb5f4d38685
https://travis-ci.org/jeysal/ngx-input-flow/builds/289622888
and exact
https://github.com/jeysal/ngx-input-flow/commit/4c7bd4ada639241e8c849ebeadead7735f3364b5
https://travis-ci.org/jeysal/ngx-input-flow/builds/289623312
versions.
Because of the behavior mentioned above, I would guess that greenkeeper-lockfile previously installed an exact version in all cases (including tilde ranges as well as carets), regardless of what `package.json` said. This matches my experience with its exact version lockfile commits for dependencies with caret versions.

Reply:
Sounds all good. Could you think of anything else where this change would break compatibility with older yarn versions?

Reply:
Not really - this is how I've always installed packages on the yarn cli myself (in those cases where I specified a version and did not just let it pick the latest) and I'd consider it the most natural option. I didn't even know about `--tilde` and `--exact` until looking at this piece of code.

Reply:
Heh, thanks, I was specifically thinking about users other than you that might have different configurations and/or habits, and/or still using older versions of yarn. I just wanna make sure we don’t start a game of whackamole where we fix something for one group of users by breaking it for others :)

Reply:
I don't think there's a yarn version that does not support this package spec syntax, it's pretty standard and has been supported in npm for a long time.
The `save-prefix` config option shouldn't matter because we specify a version range manually instead of just a package name.

I cannot imagine a case where this method could misbehave, it really just uses the range prefix that is specified in the `package.json`.